### PR TITLE
Yaling - L145.water.demand.municipal

### PR DESCRIPTION
**Pull request by yalingliupnnl**
#312

### CODE REVIEW

**Comment by bpbond** [comment]
Hi Yaling, note that the `add_precursors` block is identical for all three of this chunk's outputs. But this isn't right; they depend on _different_ inputs, and we want to track that. I'll fix but please keep in mind for next time.

**Comment by bpbond** [comment]
The `L145.municipal_water_R_W_Yh_km3` output requires a FLAG_SUM_TEST to pass. Outputs are identical however:

```
> summary(comb)
 GCAM_region_ID   water_type             year        old.value          value       
 Min.   : 1.00   Length:1280        Min.   :1971   Min.   : 0.001   Min.   : 0.001  
 1st Qu.: 8.75   Class :character   1st Qu.:1981   1st Qu.: 0.930   1st Qu.: 0.930  
 Median :16.50   Mode  :character   Median :1990   Median : 2.970   Median : 2.970  
 Mean   :16.50                      Mean   :1990   Mean   : 7.863   Mean   : 7.863  
 3rd Qu.:24.25                      3rd Qu.:2000   3rd Qu.: 9.600   3rd Qu.: 9.600  
 Max.   :32.00                      Max.   :2010   Max.   :76.600   Max.   :76.600  
> summary(comb$value - comb$old.value)
      Min.    1st Qu.     Median       Mean    3rd Qu.       Max. 
-4.441e-14  0.000e+00  0.000e+00  5.602e-17  0.000e+00  4.263e-14 
```

**Comment by yalingliupnnl** [comment]
The test package now is passed, thanks Ben!

**Comment by yalingliupnnl** [comment]
@bpbond Ben, could you help to check the NAMESPACE conflict shown above?

**Comment by bpbond** [comment]
@yalingliupnnl yes. Let's get a review from @pralitp and then deal with everything at once.

**Comment by pralitp** [comment]
Ok, all of the code in this PR is good to go.  The datasets are all documented.  However there are a number of science issues with the fundamental strategy of the code (data sets used are too incomplete to be reliable).  Some of which had already been tagged in our JIRA system (issue 212).  I'm going to go ahead and create a duplicative issue on this repository just to make sure we don't loose track of it.

**Comment by bpbond** [comment]
@pralitp Great. Thanks.

File: R/zchunk_L145.water.demand.municipal.R [hunk 1]
```diff
@@ -40,61 +42,110 @@ module_water_L145.water.demand.municipal_DISABLED <- function(command, ...) {
...
+      mutate(input.cost = expenditure / consumption) %>%
+      select(GCAM_region_ID, input.cost)->
+      municipal_water_cost_R_75USD_m3
+
+    # The IBNET data is incomplete and so it is very possible that we have entire GCAM regions in which none
+    # of the contained countries had any reported data.
+    stopifnot(nrow( municipal_water_cost_R_75USD_m3) == nrow(GCAM_region_names))
+
+    # Map water use efficiencies from the continent scale to GCAM regions
+    # Note names were changed to match manufacturing continent to avoid the need for an additional mapping
```
Comment (pralitp) [inline R/zchunk_L145.water.demand.municipal.R]: Created issue #542

File: R/zchunk_L145.water.demand.municipal.R [hunk 2]
```diff
@@ -40,61 +42,110 @@ module_water_L145.water.demand.municipal_DISABLED <- function(command, ...) {
...
+    # Note since the years are all over the place we will just use the average across years too.
+    # Come up with regional municipal prices from IBNET data
+    IBNET_municipal_water_cost_USDm3 %>%
+      inner_join(FAO_ctry_GCAM_region_ID, by = c("country" = "FAO_country")) %>%
+      mutate(expenditure = cost * consumption)  %>%
+      group_by(GCAM_region_ID) %>%
+      summarise(expenditure = sum(expenditure), consumption = sum(consumption)) %>%
+      mutate(input.cost = expenditure / consumption) %>%
+      select(GCAM_region_ID, input.cost)->
+      municipal_water_cost_R_75USD_m3
```
Comment (pralitp) [inline R/zchunk_L145.water.demand.municipal.R]: Created issue #543

**Pull request merged**